### DISCUSSION TITLE
Strange Win 11 NetGroup issue

### DISCUSSION THREAD

**Discussion by XiaoGen-Taiwan** [airsdk/Adobe-Runtime-Support#2738] [Q&A] [answered]
Hi,
This question is probably rather Windows / Windows Firewall related but just in case someone experienced the same problem ...

I have 2 apps which connects to each other via "serverless" rtmfp NetGroup.
1) Running the two apps on Windows -> no problem, they discover each other. 
2) Running the app on Android -> problem, the Android app does not discover the win clients and vice versa.
3) Running the app on macOS -> problem, the MacOS app does not discover the win clients and vice versa.
4) But the MacOS and Android clients discover each other.  

Additional Info:
- All clients connect to the same Wifi Network on my mobile phone.
- All clients receive IP's in the same range 192.168.1.XXX
- When the app opens the first time, Windows Defender Firewall shows a PopUp allowing to enable communication for that app on "Private" and / or "Public" Network.
I tried all combinations, none of them solved the problem.
- I manually added inbound and outbound rules (UDP 1935) to the Firewall ... it didn't solve the problem.
- I switched off Windows Defendere Firewall ... it didn't solve the problem.
- and the other strange thing is, that a few days ago the clients discovered themselves on all devices ...

Following an exemplary code with connection parameters:

```
var groupSpec:GroupSpecifier = new GroupSpecifier("localDiscovery");
groupSpec.postingEnabled = true;
groupSpec.routingEnabled = true;
groupSpec.serverChannelEnabled = false;
groupSpec.objectReplicationEnabled = false;
groupSpec.multicastEnabled = true;
groupSpec.peerToPeerDisabled = false;
groupSpec.ipMulticastMemberUpdatesEnabled = true;
groupSpec.addIPMulticastAddress("224.0.0.254:30000");

var netconnection:NetConnection;
var group:NetGroup;

netconnection = new NetConnection();
netconnection.addEventListener(NetStatusEvent.NET_STATUS, onNetStatus);
netconnection.connect("rtmfp:");

private function onNetStatus(evt:NetStatusEvent):void {
	switch (evt.info.code) {
		case NetStatusCode.NETCONNECTION_CONNECT_SUCCESS:
			group = new NetGroup(netConnection, groupSpec.groupspecWithoutAuthorizations());
			break;
	}
}
```

Any advice or help would be greatly appreciated

**Comment by ajwfrost** [comment, marked as answer]
Are the two Windows apps running on the same machine? or different machines?
It sounds to me like it may be related to the network discoverability/sharing settings for the public / private network. You'd need to ensure your Windows machine is set up so that your network is a Private network and you have the sharing stuff turned on (I think it is by default on Private network, and off on Public network).
If it's your own home network, then it's private, public means like wifi at a library or cafe.

On Windows 10, right-click on the wifi icon in the system tray, "Open Network & Internet Settings", and click on the "Properties" button to change from Public to Private. You can also go from here to the security settings to check the file sharing and firewall stuff..

Hope that helps!

**Reply by XiaoGen-Taiwan** [reply]
@ajwfrost , thank you soooooo much for your quick and detailed reply.

Yes, the two Windows apps were running on the same machine.

At first, when I changed the advanced sharing settings for Private Networks to "Enable Network discovery", it didn't work (Clients on different devices didn't discover each other). But since you mentioned the sharing settings I was sure that is the right way to look at .... and I found another article:
https://www.guidingtech.com/fix-network-discovery-not-working-on-windows-11/
I followed step by step but nothing solved the problem. I gave up and shutdown my Windows machine.
Then I just recalled the old times when windows has to be restarted applying a background color change.

So, I **restarted the machine** .... and baaaam it worked.

My current status of windows and firewall:
- Firewall was restored to default and no any extra rules
- No change in "services" settings (as mentioned in that article ‘Function Discovery Resource Publication’ service.
- My "Advanced sharing settings": 
Private networks -> Network discovery: ON
Private networks -> Set up network connected devices automatically: Checked
Private networks -> File and printer sharing: OFF
Public networks -> Network discovery: ON
Public networks -> Set up network connected devices automatically: Checked
Public networks -> File and printer sharing: OFF
All networks -> Public folder sharing: OFF
(as attached screenshot)
  
<img width="440" alt="Screenshot 2023-07-20 141929" src="https://github.com/airsdk/Adobe-Runtime-Support/assets/56956377/51fd8750-d77f-4426-be15-05539e57bca2">
